metadoc-core/src/main/protobuf/metadoc.proto

```diff
@@ -4,14 +4,23 @@ package metadoc.schema;
 
 message Position {
     string filename = 1;
-    int32 start = 2;
-    int32 end = 3;
+    reserved 2, 3;
+    Range range = 4;
```

Comment:
Proto 3 doesn't support required fields, since they make complicate binary forwards compatibility. I agree it's a bit annoying to drill into `references.head.range.start`. Primitives like string/int are non-optional because they fallback to default values (0, "", ...). We could simplify our life by keeping old definition and add an `.range: Range` extension method, wdyt?